package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -0,0 +1,1057 @@
+        b = set(selection[tf])
+        sort = sorted(list(a.intersection(b)))
+        return sort
+
+    def _selection_serial(self,universe,selection_str):
+        selection = []
+        for ts in universe.trajectory:
+            selection.append(universe.selectAtoms(selection_str))
+            print ts.frame
+        return selection
+    
+    # Second Legendre polynomial
+    lg2 = lambda self,x : (3*x*x - 1)/2                                                  
+
+    def run(self,**kwargs):
+        """
+        Analyze trajectory and produce timeseries
+        """
+               
+        #All the selection to an array, this way is faster than selecting later.
+        if self.nproc==1:
+            selection_out = self._selection_serial(self.universe,self.selection)
+        else:
+            #selection_out = self._selection_parallel(self.universe,self.selection,self.nproc)
+            #parallel selection to be implemented
+            selection_out = self._selection_serial(self.universe,self.selection)
+        self.timeseries = []
+        for dt in list(range(1,self.dtmax+1)):
+            output = self._getMeanOneCOHPoint(self.universe,selection_out,self.selection,dt,self.tf)
+            self.timeseries.append(output)
+
+    
+       
+class  AD:
+    r"""
+    The angular distribution function (AD) is defined as the distribution
+    probability of the cosine of the :math:`\theta` angle formed by the OH vector, HH vector
+    or dipolar vector of water molecules and a vector :math:`\hat n` parallel to chosen axis 
+    (z is the default value). The cosine is define as :math:`\cos \theta = \hat u \cdot \hat n`, where :math:`\hat u` is OH, HH or dipole vector. 
+    It creates a histogram and returns a list of lists, see Output_. The AD is also know as Angular Probability (AP).
+        
+    :Arguments:
+         *universe*
+             Universe object
+         *selection* 
+             Selection string to evaluate its angular distribution [‘byres name OH2’]
+         *bins*
+             Number of bins to create the histogram by means of numpy.histogram_ [40]
+         *axis*
+             Axis to create angle with the vector (HH, OH or dipole) and calculate cosine theta ['z']. Options: 'x',
+             'y', 'z'
+    
+    .. _numpy.histogram: http://docs.scipy.org/doc/numpy/reference/generated/numpy.histogram.html
+
+
+    """
+    def __init__(self,universe,selection_str,bins=40,nproc=1,axis="z"):
+        self.universe = universe
+        self.selection_str = selection_str
+        self.bins = bins
+        self.nproc = nproc
+        self.axis = axis
+        self.graph = None
+
+    def _getCosTheta(self,universe,selection,axis):
+        valOH = []
+        valHH = []
+        valdip= []
+        i = 0        
+        while i <= (len(selection)-1):        
+            universe.trajectory[i]
+            line = selection[ i ].positions
+            j=0
+            while j < len(line):
```

Comment:
You probably want to get rid of this loop and replace it with some big array operations..

``` python
a = line[::3]  # so this is a long column representing all the Ot0 values
b = line[1::3]
c = line[2::3]

OHVector0_s = b - a  # so this does this operation on all items in the array
HHVector0_s = c - b
```